modules/seqexec/server/src/main/scala/seqexec/server/keywords/package.scala

```diff
@@ -3,84 +3,157 @@
 
 package seqexec.server
 
+import cats.effect.IO
 import seqexec.model.dhs.ImageFileId
 
 package keywords {
+
   /**
-   * Clients that can send keywords to a server that could e.g. write them to a file
-   */
-  trait KeywordsClient {
-    def setKeywords(id: ImageFileId, keywords: KeywordBag, finalFlag: Boolean): SeqAction[Unit]
+    * Clients that can send keywords to a server that could e.g. write them to a file
+    */
+  trait KeywordsClient[F[_]] {
+
+    def setKeywords(id: ImageFileId,
+                    keywords: KeywordBag,
+                    finalFlag: Boolean): SeqActionF[F, Unit]
   }
 
-  trait DhsInstrument extends KeywordsClient {
+  trait DhsInstrument extends KeywordsClient[IO] {
     val dhsClient: DhsClient
 
     val dhsInstrumentName: String
 
-    def setKeywords(id: ImageFileId, keywords: KeywordBag, finalFlag: Boolean): SeqAction[Unit] =
+    def setKeywords(id: ImageFileId,
+                    keywords: KeywordBag,
+                    finalFlag: Boolean): SeqAction[Unit] =
       dhsClient.setKeywords(id, keywords, finalFlag)
   }
 
-  final case class StandaloneDhsClient(dhsClient: DhsClient) extends KeywordsClient {
-    override def setKeywords(id: ImageFileId, keywords: KeywordBag, finalFlag: Boolean): SeqAction[Unit] =
+  trait GDSInstrument extends KeywordsClient[IO] {
+    val gdsClient: GDSClient
+
+    def setKeywords(id: ImageFileId,
+                    keywords: KeywordBag,
+                    finalFlag: Boolean): SeqAction[Unit] =
+      gdsClient.setKeywords(id, keywords, finalFlag)
+  }
+
+  final case class StandaloneDhsClient(dhsClient: DhsClient)
+      extends KeywordsClient[IO] {
+    override def setKeywords(id: ImageFileId,
+                             keywords: KeywordBag,
+                             finalFlag: Boolean): SeqAction[Unit] =
       dhsClient.setKeywords(id, keywords, finalFlag)
   }
 
   // TODO: Implement the unsigned types, if needed.
-  sealed case class KeywordType protected (str: String)
-  object TypeInt8 extends KeywordType("INT8")
-  object TypeInt16 extends KeywordType("INT16")
-  object TypeInt32 extends KeywordType("INT32")
-  object TypeFloat extends KeywordType("FLOAT")
-  object TypeDouble extends KeywordType("DOUBLE")
-  object TypeBoolean extends KeywordType("BOOLEAN")
-  object TypeString extends KeywordType("STRING")
-
+  sealed case class KeywordType protected (str: String, gdsType: String)
+  object TypeInt8    extends KeywordType("INT8", "INT")
+  object TypeInt16   extends KeywordType("INT16", "INT")
+  object TypeInt32   extends KeywordType("INT32", "INT")
+  object TypeFloat   extends KeywordType("FLOAT", "DOUBLE")
+  object TypeDouble  extends KeywordType("DOUBLE", "DOUBLE")
+  object TypeBoolean extends KeywordType("BOOLEAN", "BOOLEAN")
+  object TypeString  extends KeywordType("STRING", "STRING")
```

Comment:
This is another instance where I let a system specific detail slip into a class that was supposed to be generic. The parameter `str` only has meaning for the DHS implementation, as the new `gdsType` only as meaning for the GDS implementation. IMHO, a better way to do this is to remove those values from `KeywordType`, and translate the types to the implementation specific values inside the DHS and GDS code.